src/components/EditProject/EditProject.js

```diff
@@ -0,0 +1,440 @@
+                          checked={selectedValue === 'a'}
+                          onChange={handleChange}
+                          value='a'
+                          color='default'
+                          name='radio-button-demo'
+                          inputProps={{ 'aria-label': 'A' }}
+                          size='small'
+                        />
+                      }
+                      label='Select'
+                      labelPlacement='start'
+                    />
+
+                    <FormControlLabel
+                      value='upload'
+                      control={
+                        <Radio
+                          checked={selectedValue === 'b'}
+                          onChange={handleChange}
+                          value='b'
+                          color='default'
+                          name='radio-button-demo'
+                          inputProps={{ 'aria-label': 'B' }}
+                          size='small'
+                        />
+                      }
+                      label='Upload'
+                      labelPlacement='start'
+                    />
+                  </Grid>
+
+                  {selectedValue == 'a' ? (
+                    <Grid container direction='row'>
+                      <Grid
+                        className={classes.gridLeft}
+                        item
+                        md={7}
+                        sm={6}
+                        container
+                        justify='center'
+                      >
+                        <SourceList
+                          onChange={setSelectSourceLanguage}
+                          width={212}
+                          value={selectSourceLanguage}
+                          componentName={'Select Source'}
+                        />
+                      </Grid>
+                      {selectSourceLanguage && (
+                        <Grid
+                          className={classes.gridLeft}
+                          item
+                          md={5}
+                          sm={6}
+                          container
+                          justify='center'
+                        >
+                          <BibleBooks
+                            onChange={setBookList}
+                            buttonText='SELECT BOOKS'
+                            sourceBooks={sourceBooks}
+                            projectBooks={projectData.metaData.books}
+                          />
+                        </Grid>
+                      )}
+                    </Grid>
+                  ) : (
+                    <Grid
+                      className={classes.gridLeft}
+                      item
+                      md={10}
+                      sm={12}
+                      container
+                      justify='center'
+                    >
+                      <UploadSource
+                        projectData={projectData}
```

Comment:
UploadSource does not have a prop of type projectData, it can take it and prevent the user from uploading previously uploaded books in the project, that check is not there, try uploading same books it fails, we can give warning before uploading in later iterations, for now we can just remove this prop.